torch/linalg/__init__.py

```diff
@@ -1029,6 +1028,13 @@
 
   - `'gelsd'` (tridiagonal reduction and SVD)
   - But if you run into memory issues: `'gelss'` (full SVD).
+  
+For CUDA input, the only valid driver is `'gels'`, which assumes that :attr:`A` is full-rank. If :attr:`A` is not square,
+the following heuristic is used to choose how the system will be solved:
+  - If the linear system to be solved is overdetermined (i.e. :attr:`A` is "tall"), 
+  then `'gels'` will solve the **least squares problem** for the system as described above.
+  - If the linear system to be solved is underdetermined (i.e. :attr:`A` is "wide" and there are infinitely many solutions),
+  then `'gels'` will find the solution which minimizes :math:`\|X\|_F`, the minimum-norm solution.
```

Comment:
This heuristic is trivial. Adding this does not add anything to the docs and makes them more convoluted. The previous way of stating things was just fine.

I'd be OK with moving the CUDA sentence here though.

Reply:
I think to solve #85021 I should explicitly mention what is done on the GPU in the underdetermined case. Maybe my most recent wording is acceptable?